python/cudf/cudf/core/series.py

```diff
@@ -6022,6 +6022,42 @@ def explode(self, ignore_index=False):
 
         return super()._explode(self._column_names[0], ignore_index)
 
+    def pct_change(
+        self, periods=1, fill_method="ffill", limit=None, freq=None
+    ):
+        """
+        Calculates the percent change between sequential elements
+        in the Series.
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Periods to shift for forming percent change.
+        fill_method : str, default 'ffill'
+            How to handle NAs before computing percent changes.
+        limit : int, optional
+            The number of consecutive NAs to fill before stopping.
+            Not yet implemented.
+        freq : str, optional
+            Increment to use from time series API.
+            Not yet implemented.
+
+        Returns
+        -------
+        Series
+        """
+        if limit is not None:
+            raise NotImplementedError("limit parameter not supported yet.")
+        if freq is not None:
+            raise NotImplementedError("freq parameter not supported yet.")
+        if fill_method not in ["ffill", "bfill"]:
```

Comment:
In pandas, the default fill method is "pad", which I believe is the same as forward fill. If we don't support `method=pad`, this may end up causing some compatibility issues in other libraries like Dask (even if it's not yet implemented in Dask)

We should verify if we can support `pad` and if so use it as the default.

Reply:
Unfortunately, `Series.fillna` only supports "ffill" and "bfill" right now. I can have `pct_change` accept "pad" and translate it into "ffill" before passing it to `fillna` if that's an acceptable solution.

Reply:
Got it. Can you file an issue for supporting pad in fillna?

Reply:
I believe the pandas codebase eventually hits this function, which indicates pad and ffill are the same and we should be able to support this. https://github.com/pandas-dev/pandas/blob/9f6a91ae785b28dc87a3cca9db3a663e2c2973fd/pandas/core/missing.py#L102

Reply:
Pad issue was filed by @TravisHester here:
https://github.com/rapidsai/cudf/issues/8727